### PR TITLE
[BUGFIX] destroyRecord should also unload

### PR DESCRIPTION
**Pull request by fivetanley**
It seems common for APIs to return a paylaod when deleting a record.
Although there is a test for when `undefined` is returned from
`deleteRecord` in the adapter here: https://github.com/emberjs/data/blob/master/tests/integration/records/delete-record-test.js#L87

There are currently no tests showing what happens when you return a
payload. It seems like the correct behavior for destroyRecord should be
`deleteRecord` + `save` + `unloadRecord`.

### CODE REVIEW

File: tests/integration/records/delete-record-test.js [hunk 1]
```diff
@@ -136,6 +136,77 @@ test('deleting a record that is part of a hasMany removes it from the hasMany re
...
+  let group;
+  let person;
+  const Group = DS.Model.extend({
+    people: DS.hasMany('person', { inverse: null, async: false })
+  });
+  Group.toString = () => { return 'Group'; }
+
+  env.adapter.deleteRecord = function() {
+    return EmberPromise.resolve({
+      data: {
```
Comment (fivetanley) [inline tests/integration/records/delete-record-test.js]: it's important to note that no matter what this value is, as long as it is not `undefined`, the test seems to still fail.

File: tests/integration/records/delete-record-test.js [hunk 2]
```diff
@@ -165,140 +216,171 @@ test('records can be deleted during record array enumeration', function(assert)
...
-    'The new person should be removed from the store'
-  );
-});
+    await record.destroyRecord();
 
-test('Destroying an invalid newly created record should remove it from the store', function(assert) {
-  var record;
-  var store = env.store;
+    assert.equal(
+      internalModel.currentState.stateName,
```
Comment (runspired) [inline tests/integration/records/delete-record-test.js]: Cc @rwjblue

**Comment by btecu** [comment]
Any progress here?

**Comment by snewcomer** [comment]
@btecu We have a few PRs here (one of which is a direct port of this PR) and are quite close.  Just need to put in the time to review.

#7226 will unblock #7258 

Hopefully going to review the native model refactor today.

**Comment by runspired** [comment]
Closing in favor of #7258

**Pull request closed without merging**